File: social-app-backend/src/main/java/razepl/dev/socialappbackend/home/data/CommentData.java
package razepl.dev.socialappbackend.home.data;

import lombok.Builder;

import java.time.LocalDate;

@Builder
public record CommentData(String commentAuthor,
                          String commentContent,
                          long commentId,
                          LocalDate commentDate,
                          long numOfLikes,
                          boolean isCommentLiked) {
}

File: social-app-frontend/src/app/core/interfaces/home/CommentData.ts
/**
 * Represents comment data.
 */
export interface CommentData {
    /**
     * The author of the comment.
     */
    commentAuthor: string;

    /**
     * The content of the comment.
     */
    commentContent: string;

    /**
     * The unique identifier of the comment.
     */
    commentId: number;

    /**
     * The date of the comment.
     */
    commentDate: Date;

    /**
     * Indicates whether the comment is liked or not.
     */
    isCommentLiked: boolean;

    /**
     * The number of likes the comment has received.
     */
    numOfLikes: number;
}
